docs/spec.yml

```diff
@@ -227,6 +227,119 @@ paths:
         '5XX':
           $ref: '#/components/responses/Unexpected'
 
+  /projects/{projectKey}/repos/{repositorySlug}/settings:
+    get:
+      summary: Get defined jobs for repository 
+      description: >
+        Returns a list of jobs defined in the
+        repository settings. Default user token omitted
+        for security.
+      parameters:
+        - name: projectKey
+          in: path
+          description: unique key for given project
+          required: true
+          schema:
+            type: string
+        - name: repositorySlug
+          in: path
+          description: unique (for project) key for given repository
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: A JSON array of server objects
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Job'
+        '404':
+          description: Project or repository not found
+        '403':
+          $ref: '#/components/responses/Unauthorized'
+        '5XX':
+          $ref: '#/components/responses/Unexpected'
+
+  /projects/{projectKey}/repos/{repositorySlug}/settings/{id}:
+    put:
+      summary: Create or update a job
+      description: >
+        Creates or updates a job in the given repository's
+        settings for the given id.
+      parameters:
+        - name: projectKey
+          in: path
+          description: unique key for given project
+          required: true
+          schema:
+            type: string
+        - name: repositorySlug
+          in: path
+          description: unique (for project) key for given repository
+          required: true
+          schema:
+            type: string
+        - name: id
+          in: path
+          description: job id
+          required: true
+          schema:
+            type: string
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Job'
```

Comment:
I didn't find, how to import defined schema without some fields (in this case we should skip `id` field as it's not needed in request body).

Alternative with separated schemas for PUT request and response view I thought it would be excess (90% info will be duplicated and will be hard to maintain).